eras/conway/impl/src/Cardano/Ledger/Conway/PParams.hs

```diff
@@ -572,6 +595,37 @@ conwayUpgradePParamsHKDPairs px pp =
   , ("dRepActivity", hkdMap px (toJSON @EpochNo) (pp ^. hkdDRepActivityL @era @f))
   ]
 
+instance ToJSON (UpgradeConwayPParams Identity) where
+  toJSON = object . upgradeConwayPParamsUpdatePairs
+  toEncoding = pairs . mconcat . upgradeConwayPParamsUpdatePairs
+
+upgradeConwayPParamsUpdatePairs :: KeyValue a => UpgradeConwayPParams Identity -> [a]
+upgradeConwayPParamsUpdatePairs upp =
+  uncurry (.=) <$> upgradeConwayPParamsHKDPairs upp
+
+upgradeConwayPParamsHKDPairs :: UpgradeConwayPParams Identity -> [(Key, Aeson.Value)]
+upgradeConwayPParamsHKDPairs UpgradeConwayPParams {..} =
+  [ ("votingThresholds", (toJSON @(Map.Map Natural Natural)) ucppVotingThresholds)
+  , ("minCCSize", (toJSON @Natural) ucppMinCCSize)
+  , ("cCTermLimit", (toJSON @Natural) ucppCCTermLimit)
```

Comment:
I made a comment on the other PR, but just in case:
```suggestion
  , ("minCommitteSize", (toJSON @Natural) ucppMinCommitteeSize)
  , ("committeeTermLimit", (toJSON @Natural) ucppCommitteeTermLimit)
```